### PR TITLE
Check all users

### PR DESCRIPTION
**Pull request by FlavioAandres**
## Checking all the uses registered in MongoDB 

The main idea is convert this to a multi tenant application. 

I'll create a new SQS to send 1 message per user to check the email, each message will trigger the lambda-checker previously created. 

## Frontend

I'll create a new section in the profile to choose the emails bank the user wants to track and the user/password of the email, maybe it would fail if the account doesn't have the properly settings to allow third party apps. 

## Updates In user document before deploying: 

**IMPORTANT: You need to set in the settings `'users.settings.email.checkedEvent' = true`  otherwise, the next checker execution will check all your last year of email notifications and you'll have duplicated data.**

## How add new fields to user: 

- Banks: there is no necessity to update directly in the database, you can use the UI 
image

- users: You need to add the email settings in your user schema, for this you can add it using the UI: 
image

All the credentials are sent to backend in base64 string and saved a hash in the database.

### CODE REVIEW

File: devutils/generateDataCreditoCredentials.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 const { encrypt } = require('../shared/utils/crypto')
 
-const username = "XXXXXXX";
-const password = "passwprd";
+const username = "flavioaandres24@gmail.com";
+const password = "vfaxynnqedzcoezh";
```
Comment (AndresMorelos) [inline devutils/generateDataCreditoCredentials.js:4]: Delete User and Password

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "aws-sdk": "^2.910.0",
```
Comment (AndresMorelos) [inline package.json:14]: AWS environment provide by default the AWS-SDK, this package is not necessary in the dependencies sections only in the dev-dependencies to perform tests in local.

**Review by AndresMorelos — CHANGES_REQUESTED** [review]
Some feedbacks given.

**Comment by AndresMorelos** [comment]
> @AndresMorelos Have you any idea if we can use the google AUTH2 to check and list emails?

@FlavioAandres yes, it is possible, here is a example provided by google https://developers.google.com/gmail/api/quickstart/nodejs

**Comment by FlavioAandres** [comment]
@AndresMorelos Hey, the scope of this PR is allow to new user to start using the app without issues, butt, I found that each user has their on bank template and if we want to replicate the bank to another account we need to create the same file to be able track the email subject and other filters... 

I want to create only one bank template and relate it into the users collection... in this way we could have an array of BankIds inside the user and only 1 template per bank... 

the other option is create that "same document" of banks for the new users  each time they were signed up and add this change of schemas in another PR :)

thoughts?

**Comment by AndresMorelos** [comment]
@FlavioAandres The best thing now is create one document per bank and then relate them to the client by id, on this form we will reduce the amount of data in the collection; and with a population of the data by id we can get the banks.


```js
//npm i mongoose-autopopulate

banks: [{
    type: mongoose.Types.ObjectId,
    ref: 'Bank',
    autopopulate: true // Needed by the plugin of autopopulation
}]

userSchema.plugin(require('mongoose-autopopulate')); // this plugin auto populate the info of the Refs
```

Then we can made a update of a user each time that they activate or desactivate a bank removing or adding the ID of the bank.

**Comment by FlavioAandres** [comment]
🤘

**Comment by FlavioAandres** [comment]
@AndresMorelos This is ready 🚀  
Deployed on my site, please go ahead merging when you deploy your changes, just to know nothing is failing :D

**Pull request merged**